go/database/mpt/io/live.go

```diff
@@ -138,6 +130,31 @@ func ExportBlockFromArchive(ctx context.Context, logger *Log, directory string,
 	return err
 }
 
+// ExportBlockFromOnlineArchive exports LiveDB view for a single given block from an Archive.
+// This method exports from the online archive, i.e, an archive that is being updated with new blocks.
+// To ensure the exported data are up-to-date, this method flushes archive to disk before exporting.
+// Expected usage is, for instance, the creation of database dump once in many blocks to backup the state.
+func ExportBlockFromOnlineArchive(ctx context.Context, logger *Log, archive *mpt.ArchiveTrie, out io.Writer, block uint64) error {
+	logger.Printf("exporting block %d from online archive", block)
+	defer func() {
+		logger.Printf("exported block %d from online archive", block)
+	}()
+
+	logger.Printf("flushing archive")
+	// before doing anything, flush the archive to ensure the data is up-to-date
+	if err := archive.Flush(); err != nil {
+		return err
+	}
+
+	logger.Printf("exporting")
+	_, err := ExportLive(ctx, logger, exportableArchiveTrie{
+		directory: archive.Directory(),
```

Comment:
It seems like the `directory` field is nowhere used but here.

Reply:
Anyway, I noticed that this directory is needed in another PR, so I will not include this change here